### PR TITLE
ARROW-16046: [Docs][FlightRPC][Python] Ensure Flight Python API is documented

### PR DESCRIPTION
**Pull request by lidavidm**
Add some missing classes to the docs, also fix a couple build warnings.

### CODE REVIEW

File: docs/source/java/vector_schema_root.rst [hunk 1]
```diff
@@ -109,7 +109,7 @@ Here is the example of building a :class:`VectorSchemaRoot`
 
 The vectors within a :class:`VectorSchemaRoot` could be loaded/unloaded via :class:`VectorLoader` and :class:`VectorUnloader`.
 :class:`VectorLoader` and :class:`VectorUnloader` handles converting between :class:`VectorSchemaRoot` and :class:`ArrowRecordBatch` (
-representation of a RecordBatch :doc:`IPC <../format/IPC.rst>` message). Examples as below
+representation of a RecordBatch :doc:`IPC <../format/IPC>` message). Examples as below
```
Comment (jorisvandenbossche) [inline docs/source/java/vector_schema_root.rst:112]: This file is no longer containing the actual contents, so pointing directly to the section in Columnar.rst might be better: `` :ref:`IPC <format-ipc>` ``

Comment (jorisvandenbossche) [inline docs/source/java/vector_schema_root.rst:112]:
```suggestion
representation of a RecordBatch :ref:`IPC <format-ipc>` message). Examples as below
```

File: python/pyarrow/_flight.pyx [hunk 2]
```diff
@@ -1168,6 +1173,10 @@ cdef class FlightClient(_Weakrefable):
     def connect(cls, location, tls_root_certs=None, cert_chain=None,
                 private_key=None, override_hostname=None,
                 disable_server_verification=None):
+        """Connect to a Flight server.
+
+        .. deprecated:: 0.15.0
```
Comment (jorisvandenbossche) [inline python/pyarrow/_flight.pyx:1178]:
```suggestion
        .. deprecated:: 0.15.0
            Use the ``FlightClient`` constructor or ``pyarrow.flight.connect`` function instead.
```

File: python/pyarrow/_flight.pyx [hunk 3]
```diff
@@ -2595,6 +2761,10 @@ cdef class FlightServerBase(_Weakrefable):
             check_flight_status(self.server.get().ServeWithSignals())
 
     def run(self):
+        """Block until the server shuts down.
+
+        .. deprecated:: 0.15.0
```
Comment (jorisvandenbossche) [inline python/pyarrow/_flight.pyx:2766]:
```suggestion
        .. deprecated:: 0.15.0
            Use the ``FlightServer.serve`` method instead
```

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

File: docs/source/java/vector_schema_root.rst (hunk 1 above)
Comment (lidavidm) [reply docs/source/java/vector_schema_root.rst]: Thanks, I gave this section a quick pass and did some copyediting in general

**Review by jorisvandenbossche — APPROVED** [review]

**Pull request closed without merging**